File: tournaments/bfsComponentSize/bfsComponentSize.py
def bfsComponentSize(matrix):
    visited = [ False for i in range(len(matrix))]
    queue = []
    componentSize = 0

    visited[1] = True
    queue.append(1)
    while len(queue) > 0:
        currentVertex = queue.pop()
        visited[currentVertex] = True
        componentSize += 1
        for nextVertex in range(len(matrix)):
            if matrix[currentVertex][nextVertex] and not visited[nextVertex]:
                visited[nextVertex] = True
                queue.append(nextVertex)

    return componentSize

File: tournaments/bfsComponentSize/bfsComponentSize.cpp
int bfsComponentSize(std::vector<std::vector<bool>> matrix) {

  std::vector<bool> visited(matrix.size(), false);
  std::queue<int> Queue;
  int componentSize = 0;

  visited[1] = true;
  Queue.push(1);

  while (Queue.size() > 0) {
    int currentVertex = Queue.front();
    Queue.pop();
    visited[currentVertex] = true;
    componentSize++;
    for (int nextVertex = 0; nextVertex < matrix.size(); nextVertex++) {
      if (matrix[currentVertex][nextVertex] && !visited[nextVertex]) {
        visited[nextVertex] = true;
        Queue.push(nextVertex);
      }
    }
  }

  return componentSize;
}
